xds/src/test/java/io/grpc/xds/SharedXdsClientPoolProviderTest.java

```diff
@@ -138,4 +157,71 @@ public void refCountedXdsClientObjectPool_getObjectCreatesNewInstanceIfAlreadySh
     assertThat(xdsClient2).isNotSameInstanceAs(xdsClient1);
     xdsClientPool.returnObject(xdsClient2);
   }
+
+  private class CallCredsServerInterceptor implements ServerInterceptor {
+    private String token;
+    private SettableFuture<Void> requestDone = SettableFuture.create();
```

Comment:
Make this `SettableFuture<String> tokenFuture` or similar. You can set it to `null` if there was no authorization header. `tokenFuture.set(metadata.get(AUTHORIZATION_METADATA_KEY));`